src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -147,6 +148,11 @@ private function populateAvailableType($id, Definition $definition)
             $this->types[$type] = $id;
         }
 
+        // Cannot use reflection if the class isn't set
```

Comment:
This can be moved above the foreach-loop as it doesn't depend on the loop.
And improves a little.